File: apps/backend-java/src/main/java/com/housingplatform/features/properties/dto/SearchPropertyCard.java
package com.housingplatform.features.properties.dto;

import com.housingplatform.persistence.enums.AccommodationType;
import java.util.List;
import java.util.UUID;

public record SearchPropertyCard(
    UUID id,
    String title,
    String slug,
    AccommodationType propertyType,
    String district,
    String nearestStationName,
    int monthlyPriceMin,
    String coverImageUrl,
    String coverImageAlt,
    List<String> tags,
    double latitude,
    double longitude,
    Double distanceMeters) {}

File: apps/backend-node/src/features/properties/dto/search-property-card.ts
import type { AccommodationType } from './accommodation-type';

export interface SearchPropertyCard {
  id: string;
  title: string;
  slug: string;
  propertyType: AccommodationType;
  district: string;
  nearestStationName: string | null;
  monthlyPriceMin: number;
  coverImageUrl: string | null;
  coverImageAlt: string | null;
  tags: string[];
  latitude: number;
  longitude: number;
  distanceMeters: number | null;
}
